python/friesian/src/bigdl/friesian/feature/table.py

```diff
@@ -2095,6 +2095,75 @@ def get_vocabularies(self, columns):
+        :param num_examples_per_list: int. The number of elements to be sampled for each list from
+               the list of each column.
+        :param random_seed: int. The number for creating 'np.random.RandomState'. Default: None.
+        :param replace: bool. Indicates whether to replace the original columns. If replace=False,
+               a corresponding column "sampled_col" will be generated for each sampled column.
+
+        :return: A new sampled listwise FeatureTable.
+        """
+        schema = self.schema
+        cols = str_to_list(columns, "cols")
+        for c in cols:
+            invalidInputError(c in self.df.columns, "Column '" + c +
+                              "' does not exist in this FeatureTable.")
+            c_type = schema[c].dataType
+            invalidInputError(isinstance(c_type, ArrayType),
+                              "Each column should be of list type, but the type of column '" + c +
+                              "' is " + c_type.simpleString())
+            if not replace:
+                c_schema = StructField("sampled_" + c, c_type, True)
+                schema.add(c_schema)
+
+        def sample_feature(row):
+            indices = row["sample_idx"]
+            sampled_feature = row.asDict()
+            for c in cols:
+                sampled_list = [sampled_feature[c][idx] for idx in indices]
+                if replace:
+                    sampled_feature[c] = sampled_list
+                else:
+                    sampled_feature["sampled_" + c] = sampled_list
+            return sampled_feature
+
+        def sample_indices(row, random_state):
+            row = row.asDict()
+            len_set = set([len(row[c]) for c in cols])
+            invalidInputError(len(len_set) == 1,
+                              "Each row of the FeatureTable should "
+                              "have the same array length in the specified cols.")
+            length = len_set.pop()
+            if length < num_examples_per_list:
+                row["sample_list"] = None
+            else:
+                sampled_indices_list = []
+                for _ in range(num_list_per_key):
+                    sampled_indices = random_state.choice(range(length), size=num_examples_per_list,
+                                                          replace=False)
+                    sampled_indices_list.append(sampled_indices.tolist())
+                row["sample_list"] = sampled_indices_list
+            return row
+
+        random_state = np.random.RandomState(seed=random_seed)
+        df = self.df.rdd.map(lambda x: sample_indices(x, random_state)).toDF()
+        df = df.dropna('any', None, subset="sample_list")
+        df = df.withColumn("sample_idx", F.explode("sample_list"))
+        spark = OrcaContext.get_spark_session()
+        df = spark.createDataFrame(df.rdd.map(lambda x: sample_feature(x)), schema)
```

Comment:
why do you need to use two udf instead of one (i.e., combining `sample_indices` and `sample_feature`)?